### PR TITLE
Allow for simpler amp-analytics tags

### LINKED ISSUE #84: Analytics won't take simple settings
### What did you expect to happen?
While changing the settings for yoast.com, I wanted to go from this:

```html
<amp-analytics type="googleanalytics">
<script type="application/json">
{
           "requests": {
               "pageviewWithCd13": "${pageview}&cd13=${cd13}"
           },
           "vars": {
               "account": "UA-random-17"
           },
           "triggers": {
               "trackPageviewWithCustom" : {
                   "on": "visible",
                   "request": "pageviewWithCd13",
                   "vars": {
                       "cd13": "AMP"
                   }                    
               }
           }
       }
</script>
</amp-analytics>
```

to this:

```html
<amp-analytics config="https://www.googletagmanager.com/amp.json?id=GTM-NTK274S&gtm.url=SOURCE_URL" data-credentials="include"></amp-analytics>
```

But when I hit save, I got an empty setting instead.

Figured out that this is due to the [aggressive sanitization here](https://github.com/Yoast/yoastseo-amp/blob/trunk/classes/class-options.php#L146).

### PR DESCRIPTION
**Pull request by jdevalk**
Fixes #84

### CODE REVIEW

File: classes/class-options.php [hunk 1]
```diff
@@ -144,51 +144,18 @@ private function sanitize_color( $color, $default ) {
...
-			$allowed_tags = strip_tags( $code, '<amp-analytics>' );
-
-			// Strip JSON content so we can apply verified script tag.
-			$tag = str_replace( $json, '', $allowed_tags );
-
-			// If the tag doesn't occur in the code, the code is invalid.
-			if ( false === strpos( $allowed_tags, '<amp-analytics' ) ) {
-				return '';
+			if ( strpos( $source, '<script type="application/json">' ) === false ) {
+				$source = strip_tags( $source, '<amp-analytics>' );
```
Comment (jcomack) [inline classes/class-options.php]: Why not immediately return?

File: classes/class-options.php [hunk 2]
```diff
@@ -261,5 +228,51 @@ private function update_post_type_settings() {
...
+				return '';
+			}
+
+			$allowed_tags = strip_tags( $code, '<amp-analytics>' );
+
+			// Strip JSON content so we can apply verified script tag.
+			$tag = str_replace( $json, '', $allowed_tags );
+
+			// If the tag doesn't occur in the code, the code is invalid.
+			if ( false === strpos( $allowed_tags, '<amp-analytics' ) ) {
```
Comment (jcomack) [inline classes/class-options.php]: Please refrain from using Yoda-style if-statements.

File: classes/class-options.php [hunk 3]
```diff
@@ -261,5 +228,51 @@ private function update_post_type_settings() {
...
+		 * Sanitizes an analytics string when it has JSON in it.
+		 *
+		 * @param string $code The code to sanitize.
+		 *
+		 * @return string Sanitized string.
+		 */
+		private function sanitize_analytics_json( $code ) {
+			// Strip all tags, to verify JSON input.
+			$json        = strip_tags( $code );
+			$parsed_json = json_decode( $json, true );
```
Comment (jcomack) [inline classes/class-options.php]: Can't this be directly called in the if-statement below? Doesn't seem to be used elsewhere (i.e. `if ( is_null( json_decode( $json, true ) ) ) { ... }`.

File: classes/class-options.php [hunk 4]
```diff
@@ -261,5 +228,51 @@ private function update_post_type_settings() {
...
+			// If the tag doesn't occur in the code, the code is invalid.
+			if ( false === strpos( $allowed_tags, '<amp-analytics' ) ) {
+				return '';
+			}
+
+			$parts    = explode( '><', $tag );
+			$parts[0] .= '>';
+			$parts[1] = '<' . $parts[1];
+
+			// Rebuild with script tag and json content.
```
Comment (jcomack) [inline classes/class-options.php]: Please capitalize json to... eh... JSON 😄

File: classes/class-options.php (hunk 2 above)
Comment (jdevalk) [reply classes/class-options.php]: Didn't touch that code :D But will change.

**Comment by jdevalk** [comment]
Tested myself with latest version just now, this fixes the bug :)

**Pull request merged**